### PR TITLE
Agda 2.4 -> 2.5.1

### PR DESCRIPTION
**Pull request by laMudri**
###### Motivation for this change

Fixed version of #15595. Doesn't modify auto-generated file, and makes all of the same intended changes.
###### Things done
- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/haskell-modules/configuration-common.nix [hunk 1]
```diff
@@ -925,10 +925,22 @@ self: super: {
...
+      files=($out/share/*-ghc-*/Agda-*/lib/prim/Agda/{Primitive.agda,Builtin/*.agda})
+      for f in "''${files[@]}" ; do
+        $out/bin/agda $f
+      done
+      for f in "''${files[@]}" ; do
+        $out/bin/agda -c --no-main $f
+      done
+      $out/bin/agda-mode compile
+    '';
+  });
```
Comment (peti) [inline pkgs/development/haskell-modules/configuration-common.nix]: It feels a little weird to have extra code in `cabal2nix` that configures a [`postInstall` hook](https://github.com/NixOS/cabal2nix/blob/master/distribution-nixpkgs/src/Distribution/Nixpkgs/Haskell/FromCabal/PostProcess.hs#L199-L205) when, apparently, that code does not work and needs to be overridden yet again. This is going to very confusing for our users.

Comment (laMudri) [inline pkgs/development/haskell-modules/configuration-common.nix]: I somewhat agree. The `postInstall` in `cabal2nix` is still okay for <2.5 versions, so it should be preserved somewhere. I guess doing a similar override on `Agda_2_4_2_3` and `Agda_2_4_2_4` in `configuration-common.nix` (and getting rid of the stuff in `cabal2nix`) would be a reasonably sensible thing to do.

Comment (peti) [inline pkgs/development/haskell-modules/configuration-common.nix]: Actually, I believe that cabal2nix is the right place for that kind of information. The override files in Nixpkgs are a kludge that shouldn't be used unless we have to. It's much nicer for everyone if cabal2nix generates correct expressions in the first place.

**Comment by laMudri** [comment]
So, how do we coördinate this? I don't know how to test any changes to cabal2nix, so I'm relying on someone else here. All commits other than the first in this pull request should be okay to merge in as they are.

**Comment by peti** [comment]
The new Agda code is in the`haskell-updates` branch already. I'll merge to `master` ASAP.

**Comment by laMudri** [comment]
Okay, thanks!

**Comment by laMudri** [comment]
Closed in favour of #16038.

**Pull request closed without merging**